src/service/db/dashboards/mod.rs

```diff
@@ -220,54 +177,36 @@ pub(crate) async fn list(org_id: &str, folder: &str) -> Result<Vec<Dashboard>, a
         .await?
         .into_values()
         .map(|val| {
-            let dash_str = std::str::from_utf8(&val)?;
-            let hash = config::utils::hash::gxhash::new()
-                .sum64(dash_str)
-                .to_string();
-            let d_version: DashboardVersion = json::from_slice(&val).unwrap();
-            if d_version.version == 1 {
-                let dash: v1::Dashboard = json::from_slice(&val).unwrap();
-                Ok(Dashboard {
-                    v1: Some(dash),
-                    version: 1,
-                    hash,
-                    ..Default::default()
-                })
-            } else if d_version.version == 2 {
-                let dash: v2::Dashboard = json::from_slice(&val).unwrap();
-                Ok(Dashboard {
-                    v2: Some(dash),
-                    version: 2,
-                    hash,
-                    ..Default::default()
-                })
-            } else if d_version.version == 3 {
-                let dash: v3::Dashboard = json::from_slice(&val).unwrap();
-                Ok(Dashboard {
-                    v3: Some(dash),
-                    version: 3,
-                    hash,
-                    ..Default::default()
-                })
-            } else if d_version.version == 4 {
-                let dash: v4::Dashboard = json::from_slice(&val).unwrap();
-                Ok(Dashboard {
-                    v4: Some(dash),
-                    version: 4,
-                    hash,
-                    ..Default::default()
-                })
+            let dash = from_json(&val)?;
+            Ok(dash)
+        })
+        .collect()
+}
+
+
+/// Searches for dashboards whose titles contain the given pattern.
+#[tracing::instrument]
+pub(crate) async fn search(org_id: &str, title_pat: Option<&str>) -> Result<Vec<Dashboard>, anyhow::Error> {
+    let db_key = format!("/dashboard/{org_id}/");
+    let ds = db::list(&db_key)
+        .await?
+        .into_values()
+        .filter_map(|val| {
+            let dash = from_json(&val).ok()?;
+            Some(dash)
+        })
+        .filter(|dash| {
+            if let Some(title_pat) = title_pat {
+                // Only include the dashboard if it has a title and the title
+                // contains the given title pattern.
+                dash.title().filter(|t| t.contains(title_pat)).is_some()
```

Comment:
We have to do this instead of the simpler

```
dash.title().contains(title_pat).is_some()
```

since `tile()` returns an `Option<&str>` instead of a `&str`.